cirq-core/cirq/protocols/measurement_key_protocol.py

```diff
@@ -290,3 +290,14 @@ def with_key_path(val: Any, path: Tuple[str, ...]):
     """
     getter = getattr(val, '_with_key_path_', None)
     return NotImplemented if getter is None else getter(path)
+
+
+def with_key_path_prefix(val: Any, path: Tuple[str, ...]):
```

Comment:
@95-martin-orion Might we want to make this private? IDK if there's a precedent for private protocols. But I don't see much realworld use case for this to be used publicly.

If so, we may also want to deprecate `with_key_path` and make that internal as well.

Reply:
Nit: prefer `prefix` (instead of `path`) for the prefix argument.

I'm not aware of a precedent for private protocols either - given the nature of protocols as something that can be used across Cirq, it seems sort of counterintuitive. In particular, if an external user creates a class that implements this protocol (e.g. a new operation with a measurement key), they would need to invoke this method to propagate its effects - calling `_with_key_path_prefix_` directly is a recipe for breakages.